src/toolbar/Toolbar.jsx

```diff
@@ -1,15 +1,254 @@
+                        break;
+                    }
+
+                    /*
+                     * The number of icons to be displayed are limited because we need to make sure that
+                     *  they will fit in the toolbar main container's width
+                     */
+                    displayIcons.push(toolbarIcons[idx].icon);
+                }
+            }
+
+            // This will determine if we need to display the left arrow icon
+            if (this.state.startIconIndex > 0) {
+                displayArrowLeft = (
+                    <div className="chamel-toolbar-left-arrow">
+                        <FontIcon onClick={this._handleArrowClick.bind(this, -1)}
+                                  className="cfi cfi-chevron-left"/>
+                    </div>
+                );
+            }
+
+            // This will determine if we need to display the right arrow icon
+            if ((this.state.startIconIndex + displayIcons.length) < toolbarIcons.length) {
+                displayArrowRight = (
+                    <div className="chamel-toolbar-right-arrow">
+                        <FontIcon onClick={this._handleArrowClick.bind(this, 1)}
+                                  className="cfi cfi-chevron-right"/>
+                    </div>
+                );
+            }
+
+            // Modify the toolbar main container's style so we can properly display the toolbar icons
+            let styleToolbarMainCon = {
+                position: 'absolute',
+                left: DEFAULTICONWIDTH + 'px',
+                width: this.state.toolbarContainerWidth + 'px'
+            };
+
+            return (
+                <div ref="chamelToolbar" className="chamel-toolbar" style={{padding: '0px'}}>
+                    {displayArrowLeft}
+                    <div className="chamel-toolbar-group" style={styleToolbarMainCon}>
+                        {displayIcons}
+                    </div>
+                    {displayArrowRight}
+                </div>
+            );
+        } else {
+
+            // If the toolbar main container is wide enough to display all the icons, then just display the props.children
+            return (
+                <div ref="chamelToolbar" className="chamel-toolbar">
+                    {this.props.children}
+                </div>
+            );
+        }
+    },
+
+    /**
+     * Function that will get all the possible toolbar icons
+     *
+     * @param {React.Children} children The children of the react component that contains the possible toolbar icons
+     * @private
+     */
+    _getToolbarIcons: function (children) {
+
+        // Map thru the react children and evaluate each element if it is a toolbar icon or a parent component
+        React.Children.map(children, function (element) {
+
+            // Evaluate the element type if it is a toolbar icon
+            switch (element.type.displayName) {
```

Comment:
This is throwing an error for me: Uncaught TypeError: Cannot read property 'displayName' of undefined